### PR TITLE
add article vote functionality

### PR DESCRIPTION
**Pull request by rorybush**

### CODE REVIEW

File: src/Components/SingleArticle.js [hunk 1]
```diff
@@ -8,12 +8,25 @@ function SingleArticle() {
...
 
   const [SingleArticle, setSingleArticle] = useState({});
+  const [VoteError, setVoteError] = useState("");
 
   useEffect(() => {
     api.fetchArticleById(article_id).then(({ article }) => {
       setSingleArticle(article);
     });
-  }, [article_id]);
+  }, [article_id, SingleArticle]);
```
Comment (B7Bprog) [inline src/Components/SingleArticle.js:17]: Using SingleArticle here causes an infinite re-render. Please remove it from the dependency array.

File: src/Components/SingleArticle.js [hunk 2]
```diff
@@ -8,12 +8,25 @@ function SingleArticle() {
...
+
+  const handleArticleUpVote = () => {
+    api.updateArticleVotes(article_id, 1).catch((err) => {
```
Comment (B7Bprog) [inline src/Components/SingleArticle.js:20]: Implement optimistic rendering here and in 'DownVote'.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Rory, currently you've got an infinite loop going on due to the wrong dependency in your useEffect. (See my comment). Please refactor that and also make sure you apply optimistic rendering as well at the same time with the API request.

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
The app is looking good, things are coming along really nicely now! There's a few things I think you could build upon with this ticket to really polish things off:
- Optimistic rendering of the votes, currently if you throttle the network to 'slow 3G' and click upvote or downvote nothing happens for a few seconds, try to introduce optimistic rendering so that the vote count changes instantly and then the patch request is made in the background 👍 This provides the user with a much snappier and impactful experience
- There isn't a way to navigate back to the list of articles currently from the single article page ( outside of using the back button )
- If you can I would try to lean away from displaying info in key-value form such as `comments: 11, votes: 9` - generally speaking you don't need labels as often as you think, try to lean more towards `11 Comments, 9 Votes` as it feels less blocky

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hi Rory, optimistic rendering seems to be well sorted. Please check the home button on the single article page, as it doesn't seem to navigate back to all the articles/ home. Also you should display a message ('Loading...') while the single article is loading. 
Loading at comments is fine. 

Please amend your code and then feel free to merge.

**Pull request merged**